src/campaign_server/addon_utils.cpp

```diff
@@ -92,23 +92,43 @@ std::string format_addon_feedback_url(const std::string& format, const config& p
 	return std::string();
 }
 
+void support_translation(config& addon, std::string locale_id)
+{
+	if(!locale_id.empty()) {
+		config& locale = addon.find_child("translation", "language", locale_id);
+		if(locale) {
+			locale["supported"] = true;
+		}
+		else {
+			addon.add_child("translation")["language"] = locale_id; //Doing this to circumvent a weird validation bug
+			addon.find_child("translation", "language", locale_id)["supported"] = true;
+		}
+	}
+}
+
```

Comment:
A couple of style issues:
* `locale_id` should be a const reference here. The check for its emptiness makes up most of the function body instead of returning early if it passes.
* The `else` branch performs excessive WML scans. What's going on here is that you need to create the WML node in line 103 (`add_child()`), take its reference, and then do all further operations (adding the `language` and `supported` attribute values) with that reference.
* Based on that, the comment in there is also redundant. The validation bug you were hitting is the infamous "missing WML child yet unchecked for" error — if `find_child()` does not actually find the node you were looking for it returns a dummy WML node that can't be used in any meaningful way beyond checking its validity (the `bool` conversion). Since you already figured that out and are creating the child accordingly, you don't need to point it out.

Reply:
1. Very well.
2,3. Initially, I tried
```
locale = addon.add_child("translation");
locale["language"] = locale_id;
locale["supported"] = true;
```

But it crashed on the first line due to the validation error because the `=` operator had been overloaded with a validation check.
I understand from your suggestion that I should try referring it with a pointer instead? 
Something like this: `locale_pointer = &addon.add_child("translation");` ?

Reply:
I'll add the emptiness check to the sanitation check, then.

Reply:
The reason you can't assign to `locale` if the WML node is absent is that in C++ references are immutable. Once they are initialized, they are pointing to whatever they were initialized with (even if it's destroyed, this creates a dangling reference, which leads to undefined behaviour). So you need to take a new reference from the return value of `add_child()` -- otherwise you'd be attempting to call the assignment operator for the original (dummy) instance with the result of `add_child()` as its argument.